linkcheck/director/task.py

```diff
@@ -13,10 +13,8 @@
 # You should have received a copy of the GNU General Public License along
 # with this program; if not, write to the Free Software Foundation, Inc.,
 # 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
-try: # Python 3
-    import _thread
-except ImportError: # Python 2
-    import thread as _thread
+import _thread
```

Comment:
Another `interrupt_main()`.

TBH this business of catching KeyboardInterrupt in threads and calling _thread.interrupt_main()?  I could've sworn that only the main thread in a Python process ever gets KeyboardInterrupt (because only the main thread handles signals), but https://docs.python.org/3/library/_thread.html claims otherwise:

> Threads interact strangely with interrupts: the KeyboardInterrupt exception will be received by an arbitrary thread. (When the signal module is available, interrupts always go to the main thread.)

Then again, when is the `signal` module not available?  It's even available on Windows!

So I don't know.  I'd like to dig into git history and see if there's anything about adding this code in response to a user's bug report, or if it was added purely theoretically.

IOW out of scope for this PR, let's leave it as is for now.

Reply:
Raised  386.